src/main/scala/org/broadinstitute/dsde/workbench/leonardo/api/LeoRoutes.scala

```diff
@@ -0,0 +1,77 @@
+package org.broadinstitute.dsde.workbench.leonardo.api
+
+import akka.actor.ActorSystem
+import akka.event.Logging.LogLevel
+import akka.event.{Logging, LoggingAdapter}
+import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport
+import SprayJsonSupport._
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.server
+import akka.http.scaladsl.server.Directives.{complete, get, handleExceptions, pathEndOrSingleSlash, pathPrefix}
+import akka.http.scaladsl.server.{Directive0, ExceptionHandler}
+import akka.http.scaladsl.server.RouteResult.Complete
+import akka.http.scaladsl.server.directives.{DebuggingDirectives, LogEntry, LoggingMagnet}
+import akka.stream.Materializer
+import akka.stream.scaladsl.Sink
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.workbench.leonardo.model.ErrorReport
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class LeoRoutes()(implicit val system: ActorSystem, val materializer: Materializer, val executionContext: ExecutionContext)
+  extends LazyLogging {
+
+  def leoRoutes: server.Route =
+    pathPrefix("ping") {
+      pathEndOrSingleSlash {
+        get {
+          complete {
+            StatusCodes.OK
+          }
+        }
+      }
+    }
+
+  def route: server.Route = (logRequestResult & handleExceptions(myExceptionHandler)) {
+    leoRoutes
+  }
+
+  private val myExceptionHandler = {
+    import org.broadinstitute.dsde.workbench.leonardo.model.ErrorReportJsonSupport._
+
+    ExceptionHandler {
+      //case withErrorReport: WorkbenchExceptionWithErrorReport =>
+      //  complete(withErrorReport.errorReport.statusCode.getOrElse(StatusCodes.InternalServerError), withErrorReport.errorReport)
+      case e: Throwable =>
+        complete(StatusCodes.InternalServerError, ErrorReport(e))
+    }
+  }
+
+  // basis for logRequestResult lifted from http://stackoverflow.com/questions/32475471/how-does-one-log-akka-http-client-requests
```

Comment:
i stole this from SAM. i could read the stackoverflow link and summarise it into a comment, but so could the reader ;)